src/interface/viewport.c

```diff
@@ -768,7 +768,7 @@ void viewport_paint(rct_viewport* viewport, rct_drawpixelinfo* dpi, int left, in
 		paint_quadrant_ps();
 
 		int weather_colour = WeatherColours[gClimateCurrentWeatherGloom];
-		if ((weather_colour != -1) && (!(gCurrentViewportFlags & VIEWPORT_FLAG_INVISIBLE_SPRITES)) && (!(RCT2_GLOBAL(0x9DEA6F, uint8) & 1))){
+		if ((weather_colour != -1) && (!(gCurrentViewportFlags & VIEWPORT_FLAG_INVISIBLE_SPRITES)) && (!(RCT2_GLOBAL(0x9DEA6F, uint8) & 1)) && gConfigGeneral.render_weather_gloom){
```

Comment:
This if statement would better be spread out over multiple lines to increase legibility.